aws/iam/instance_profile/main.tf

```diff
@@ -24,6 +24,15 @@ resource "aws_iam_role" "mod" {
   path = "/"
 }
 
+data "aws_iam_policy" "AmazonSSMManagedInstanceCore" {
+  arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+}
+
+resource "aws_iam_role_policy_attachment" "mod-ssm-role-policy" {
+  role       = aws_iam_role.mod.id
+  policy_arn = data.aws_iam_policy.AmazonSSMManagedInstanceCore.arn
```

Comment:
why do you need the policy document if you are passing the ARN?

Reply:
the policy document above is just a reference to the aws one. We could just pass the arn direct, i think, but this puts a little more structure and validation around it. I have no strong feelings on this one, it was just what I researched this morning for using AWS managed policies in terraform :)